mne/io/nirx/nirx.py

```diff
@@ -0,0 +1,204 @@
+        hdr = cp.RawConfigParser()
+        hdr.read_string(hdr_str)
+
+        # Check that the file format version is supported
+        assert (hdr['GeneralInfo']['NIRStar'] == "\"15.2\""), \
+            "Only NIRStar version 15.2 is supported"
+
+        # Parse required header fields
+
+        # Extract source-detectors requested by user
+        sources = [int(s) for s in re.findall(r'(\d)-\d:\d+',
+                   hdr['DataStructure']['S-D-Key'])]
+        detectors = [int(s) for s in re.findall(r'\d-(\d):\d+',
+                     hdr['DataStructure']['S-D-Key'])]
+        sdindex = [int(s) for s in re.findall(r'\d-\d:(\d+)',
+                   hdr['DataStructure']['S-D-Key'])]
+        assert (len(sources) == len(detectors)), \
+            "Same amount of sources and detectors required"
+        assert (len(sources) == len(sdindex)), \
+            "Same amount of sources and keys required"
+
+        # Create mne structure
+        # TODO: ch_type is currently misc as I could not find appropriate
+        #       other type, the hbo and hbr type are not relevant until the
+        #       signal has been converted
+        # TODO: nchan needs to be multiplied by two as we have two wavelengths
+        #       per sensor should the underlying type be modified to
+        #       support (wavelength x channels x data)?
+        info = create_info(len(sources) * 2,
+                           hdr['ImagingParameters']['SamplingRate'],
+                           ch_types='misc')
+        info.update({'subject_info': subject_info})
+
+        # Store the subset of sources and detectors requested by user
+        # The signals between all source-detectors are stored even if they
+        # are meaningless, the user pre specifies which combinations are
+        # meaningful
+        # TODO: Is this style of info overloading allowed?
+        #       Currently I am marking all things as fnirs_ to make easier to
+        #       find and remove if there is more appropriate storage locations
+        info['fnirs_sources'] = np.asarray(sources)
+        info['fnirs_detectors'] = np.asarray(detectors)
+        info['fnirs_sdindex'] = np.asarray(sdindex)
+
+        fnirs_wavelengths = [int(s) for s in
+                             re.findall(r'(\d+)',
+                             hdr['ImagingParameters']['Wavelengths'])]
+        info['fnirs_wavelengths'] = np.asarray(fnirs_wavelengths)
+
+        super(RawNIRX, self).__init__(
+            info, preload, filenames=[fname], last_samps=[last_sample],
+            raw_extras=[hdr], verbose=verbose)
+
+    def _read_segment_file(self, data, idx, fi, start, stop, cals, mult):
+        """Read a segment of data from a file.
+        The NIRX machine records raw data as two different wavelengths.
+        These are stored in two files [wl1, wl2]. This function will
+        return the data as ([wl1, wl2] x samples)
+        """
+
+        file_wl1 = glob.glob(self.filenames[fi] + '/*.wl1')
```

Comment:
This is the sort of thing that you'd want in `raw_extras`. Instead of re-globbing all the time you could just save the `files` dict in the `self._raw_extras[fi]` entry.